File: analytics-firebase/src/test/java/com/mindera/skeletoid/analytics/appenders/FBAppenderUnitTest.java
package com.mindera.skeletoid.analytics.appenders;

import static junit.framework.Assert.assertNotNull;
import static org.mockito.Mockito.mock;

/**
 * Created by Pedro Vicente - pedro.vicente@mindera.com
 * File created on 12/11/2018.
 */
public class FBAppenderUnitTest {

    private static final String mPackageName = "my.package.name";

    private android.content.Context mContext;

    @org.junit.Before
    public void setUp() {
        mContext = mock(android.content.Context.class);
    }

    @org.junit.Test
    public void testFBNotNull() {
        FBAppender fb = new FBAppender();

        fb.disableAppender();

        assertNotNull(fb);


    }
}

File: analytics-firebase/src/test/java/com/mindera/skeletoid/analytics/appenders/FBAppenderUnitTest.kt
package com.mindera.skeletoid.analytics.appenders

import android.content.Context

import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

import org.mockito.Mockito.mock
import org.mockito.Mockito.times
import org.mockito.Mockito.verify

@RunWith(RobolectricTestRunner::class)
@Config(manifest = Config.NONE)
class FBAppenderUnitTest {

    private var mContext: Context? = null

    @Before
    fun setUp() {
        mContext = mock<Context>(Context::class.java)
    }

    @Test
    fun testDisableAppender() {
        val appender = FBAppender()
        appender.disableAppender()

        verify(appender, times(1)).setUserProperty("mFirebaseAnalytics", null)
    }
}
